### DISCUSSION TITLE
CSLA 8 prerelease 24032503 online

### DISCUSSION THREAD

**Discussion by rockfordlhotka** [MarimerLLC/csla#3750] [Announcements]
I have pushed CSLA 8 prerelease 24032503 to NuGet.

This includes fixes to the Blazor 8 state management subsystem, including some changes that may break code from previous prereleases.

Most notably:

* The default is now to NOT use the new state management, so a server-side Blazor app won't incur overhead, all state is in DI
* If you use a mix of render modes (server-interactive and server-static, or server anything with wasm-interactive) you'll need to do the work to enable the new Blazor 8 state management
* ProjectTracker now demonstrates the use of (and configuration of) the new state management
* The new new state management (as of this prerelease) no longer makes any attempt to "lock" session state, so it is up to you to avoid overwriting your own state
   * When you have pure server side pages (interactive, static, streaming) they should all play together without worry - if you have only one web server
   * When you have pure wasm side pages they should all play together without worry
   * When you mix server and wasm pages with the new state management, you _must_ call `StateManager.SaveState` to update the state management subsystem - this is true for server and wasm pages, because this method updates a change-stamp so other consumers of the state know they need to refresh the data

This is all the result of issue #3740 identified by @Tony20221 (thank you!!)

**Comment by uwefms** [comment]
Hi Rocky,
  
    thank you for your great work. I downloaded the new code version, but the ProjectTracker sample seems to be unchanged resulting in an error in the get section of CslaStateController.cs due to a change in the base class of the new version. Am I too impatient or did you forget to push the ProjectTracker changes (as promised in the upper text) or is there another source I can reach.

    regards
    
         Uwe

**Reply by rockfordlhotka** [reply]
The latest ProjectTracker is still in the PR waiting for review.

**Reply by rockfordlhotka** [reply]
I merged the PR, so ProjectTracker in `main` should be current.
